### PR TITLE
Display badge count

### PR DESCRIPTION
**Pull request by amit-mb**
Displays the number of listened requests as Badge text

### CODE REVIEW

File: app/background.ts [hunk 1]
```diff
@@ -59,6 +67,7 @@ class BackgroundWorker {
     const tabRecords = this.data[this.currentTab];
     if (tabRecords.enabled && this.currentTab === details.tabId) {
       tabRecords.requests.push(details);
+      tabRecords.count += 1
```
Comment (revathskumar) [inline app/background.ts]: why keep this separate? just take the `tabRecords.requests.length`.

File: app/popup.tsx [hunk 2]
```diff
@@ -103,6 +103,13 @@ chrome.tabs.query(queryParams, tabs => {
   const { id, url } = tab;
   if (typeof id === "undefined" || typeof url === "undefined") return;
 
+  MessageService.getCount(id, (count) => {
+  chrome.browserAction.setBadgeText({
+      text: `${count}`,
```
Comment (revathskumar) [inline app/popup.tsx]: no need for the string interpolation.
just use `text: count`

File: app/background.ts [hunk 3]
```diff
@@ -51,6 +55,10 @@ class BackgroundWorker {
           sendResponse(this.data[this.currentTab].requests);
           break;
         }
+        case 'GET_COUNT' : {
+          sendResponse(this.data[this.currentTab].count);
```
Comment (revathskumar) [inline app/background.ts]: `this.data[this.currentTab].requests.length`

File: app/background.ts [hunk 4]
```diff
@@ -23,6 +23,10 @@ class BackgroundWorker {
   }
 
   startMessageListener() {
+    chrome.browserAction.setBadgeText({
+      text: `${this.data[this.currentTab].count}`,
```
Comment (revathskumar) [inline app/background.ts]: No need for string interpolation.

File: app/background.ts (hunk 1 above)
Comment (amit-mb) [reply app/background.ts]: Yeah, thats better

File: app/popup.tsx (hunk 2 above)
Comment (amit-mb) [reply app/popup.tsx]: We have declared count to be of number type https://github.com/code-mancers/interceptor/blob/4cda610db7930ff9c4042e055adc5799615f118c/app/background.ts#L8 , we need interpolation I guess.

**Pull request merged**